nomad/node_endpoint.go

```diff
@@ -1140,50 +1142,75 @@ func (n *Node) UpdateAlloc(args *structs.AllocUpdateRequest, reply *structs.Gene
 		}
 
 		// if the job has been purged, this will always return error
-		job, err := n.srv.State().JobByID(nil, alloc.Namespace, alloc.JobID)
+		var job *structs.Job
+		var jobType string
+		var jobPriority int
+
+		job, err = n.srv.State().JobByID(nil, alloc.Namespace, alloc.JobID)
 		if err != nil {
 			n.logger.Debug("UpdateAlloc unable to find job", "job", alloc.JobID, "error", err)
-			continue
```

Comment:
The only cases we can throw an error are all pretty bad (ex. invalid keys, txn failures). I'm not sure we should try to keep going in this case.

Reply:
I think the orphaned alloc case is only where `JobByID` returns nil for job (which you're handling below)? Getting an error from `JobByID` means something has gone really wrong (mostly in go-memdb code), not that there wasn't a result.